tutorial_src/gue/prog/gue_lib.cc

```diff
@@ -0,0 +1,107 @@
+#include <iostream>
+
+#include "gue.h"
+
+using namespace std;
+
+namespace {
+
+/*
+ * IN:
+ * N (^143)
+ * X (^N)
+ *
+ * OUT:
+ * magic line
+ * OK/WRONG
+ * queries
+ * library comment
+ */
+
+int N, X;
+
+const string magicLine = "GoodJobYouDiDEndTheGame!!";
+
+// uper and lower bounds of interval of possible answers
+int upperBound, lowerBound;
+
+bool initDone = false, ansDone = false;
+int queries = 0;
+
+void output(bool correct, string comment) {
+    cout << magicLine << "\n";
+    cout << (correct ? "OK" : "WRONG") << "\n";
+    cout << queries << "\n";
+    cout << comment << "\n";
+}
+
+void assertCondition(bool cond, string err_msg) {
+    // można zauważyć, że kontrukcja sprawdzaczki pozwala wypisać wiele razy
+    // ona złapie tylko pierwszy błąd
+    if (!cond)
+        output(false, err_msg);
+}
+
+void check() {
+    assertCondition(initDone && !ansDone, "Querries order is incorrect");
```

Comment:
QueRies, jedno R.
A w zasadzie "Order of queries"

Ciekawostka językowa: querry też istnieje, ale oznacza co innego: https://en.wiktionary.org/wiki/querry